### PR TITLE
Feature: Add printing worklfow optimize

### PR DESCRIPTION
**Pull request by IDKHTS**
- comfirm modal before stop and pause job
- global setting: hide console 
- preview modal show gcodeFile params

### CODE REVIEW

File: src/app/resources/i18n/en/resource.json [hunk 1]
```diff
@@ -1195,6 +1197,23 @@
     "key-Workspace/WifiTransport-Sending File": "Send to Device",
     "key-Workspace/WifiTransport-Sending file. Please wait…": "Sending file. Please wait…",
     "key-Workspace/WifiTransport-Your file was successfully sent. Receive it on the Touchscreen.": "Your file was successfully sent. Receive it on the Touchscreen.",
+    
+    "key-Workspace/GCodeParams-GCode Type": "G-code 类型",
+    "key-Workspace/GCodeParams-Estimated time": "预估时间",
```
Comment (jane-rose) [inline src/app/resources/i18n/en/resource.json:1202]: use english

File: src/app/ui/widgets/Console/Console.jsx [hunk 2]
```diff
@@ -281,16 +282,15 @@ function Console({ widgetId, widgetActions, minimized, isDefault, clearRenderSta
     }, [isConnected, port, server]);
 
     useEffect(() => {
-        if (connectionType === CONNECTION_TYPE_SERIAL) {
+        const wifiWorking = workflowStatus === WORKFLOW_STATUS_RUNNING || workflowStatus === WORKFLOW_STATUS_PAUSED;
+        const serialWoking = workflowState === WORKFLOW_STATE_RUNNING || workflowState === WORKFLOW_STATE_PAUSED;
```
Comment (jane-rose) [inline src/app/ui/widgets/Console/Console.jsx:286]: have to rebase

File: src/app/ui/widgets/WifiTransport/WifiTransport.jsx [hunk 3]
```diff
@@ -312,6 +318,13 @@ function WifiTransport({ widgetActions, controlActions }) {
                 actions.loadGcodeToWorkspace();
             }
         },
+        matchingGCodeTypeAndHeadType: async () => {
```
Comment (jane-rose) [inline src/app/ui/widgets/WifiTransport/WifiTransport.jsx:321]: add await?

File: src/app/ui/widgets/WorkingProgress/Loading.jsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Spin } from 'antd';
+import { LoadingOutlined } from '@ant-design/icons';
+
+// const Loading = (props) => <LoadingOutlined {...props} style={{ fontSize: 19, ...props.style }} spin />;
```
Comment (jane-rose) [inline src/app/ui/widgets/WorkingProgress/Loading.jsx:7]: remove log

File: src/app/ui/widgets/WorkingProgress/WorkingProgress.jsx [hunk 5]
```diff
@@ -20,6 +22,51 @@ Text.propTypes = {
...
+            visible
+            onClose={() => { props.onClose(); }}
+        >
+            <Modal.Header>
+                停止作业
+                {/* {i18n._(headerTextKey)} */}
+            </Modal.Header>
+            <Modal.Body>
+                {props.children}
+                你确定要停止打印作业吗？
```
Comment (jane-rose) [inline src/app/ui/widgets/WorkingProgress/WorkingProgress.jsx:38]: use english

File: src/app/ui/widgets/WorkingProgress/WorkingProgress.jsx [hunk 6]
```diff
@@ -30,13 +30,13 @@ const StopConfirmModal = (props) => {
...
-                停止作业
-                {/* {i18n._(headerTextKey)} */}
+                {/* 停止作业 */}
+                {i18n._('key-Workspace/Workprogress-StopJobConfirmModal title')}
             </Modal.Header>
             <Modal.Body>
                 {props.children}
-                你确定要停止打印作业吗？
-                {/* {i18n._(bodyTextKey)} */}
+                {/* 你确定要停止打印作业吗？ */}
```
Comment (jane-rose) [inline src/app/ui/widgets/WorkingProgress/WorkingProgress.jsx:39]: please remove this line

**Review by jane-rose — APPROVED** [review]

**Review by xiekun1992 — APPROVED** [review]

**Pull request merged**